assets/global.js

```diff
@@ -553,6 +564,131 @@ class SliderComponent extends HTMLElement {
+    this.sliderControlLinksArray = Array.from(this.sliderControlLinks);
+    this.sliderAutoplay = this.slider.hasAttribute('data-autoplay');
+    this.mediaQueryMobile = window.matchMedia('(max-width: 749px)');
+    this.isPlaying = false;
+    this.autoPlayEnabled = false;
+
+    const resizeObserverControls = new ResizeObserver(entries => this.styleSlideshowControls());
+    resizeObserverControls.observe(this);
+    this.sliderControlLinks.forEach(link => link.addEventListener('click', this.linkToSlide.bind(this)));
+    this.slider.addEventListener('scroll', this.setSlideVisibility.bind(this));
+    this.setSlideVisibility();
+
+    if (!this.sliderAutoplay) return;
+    this.sliderAutoplayButton = this.sliderControlWrapper.querySelector('.slideshow__autoplay');
+    this.autoplaySpeed = this.slider.dataset.speed * 1000;
+
+    this.sliderAutoplayButton.addEventListener('click', this.autoPlayToggle.bind(this));
+    this.slider.addEventListener('mouseenter', this.autoplayFocusHandling.bind(this));
+    this.slider.addEventListener('mouseleave', this.autoplayFocusHandling.bind(this));
+    this.slider.addEventListener('focusin', this.autoplayFocusHandling.bind(this));
+    this.slider.addEventListener('focusout', this.autoplayFocusHandling.bind(this));
+    this.prevButton.addEventListener('focusout', this.autoplayFocusHandling.bind(this));
+    this.prevButton.addEventListener('focusin', this.autoplayFocusHandling.bind(this));
+    this.nextButton.addEventListener('focusin', this.autoplayFocusHandling.bind(this));
+    this.nextButton.addEventListener('focusout', this.autoplayFocusHandling.bind(this));
+    this.prevButton.addEventListener('mouseenter', this.autoplayFocusHandling.bind(this));
+    this.prevButton.addEventListener('mouseleave', this.autoplayFocusHandling.bind(this));
+    this.nextButton.addEventListener('mouseenter', this.autoplayFocusHandling.bind(this));
+    this.nextButton.addEventListener('mouseleave', this.autoplayFocusHandling.bind(this));
+    this.play();
+  }
+
+  autoPlayToggle() {
+    this.isPlaying ? this.pause() : this.play();
+  }
+
+  autoplayFocusHandling(event) {
+    const shouldPause = event.type === 'mouseenter' || event.type === 'focusin';
+    const shouldPlay = event.type === 'mouseleave' || event.type === 'focusout';
+    if (this.isPlaying && shouldPause) {
+      this.wasPlaying = true;
+      this.pause();
+    } else if (this.wasPlaying && shouldPlay) {
+      this.play();
+    } else {
+      this.wasPlaying = false;
+    }
+  }
+
+  play() {
+    if (this.isPlaying) return;
+
+    this.isPlaying = true;
+    this.sliderAutoplayButton.classList.remove('slideshow__autoplay--paused');
+    this.slider.setAttribute('aria-live', 'off');
+    this.sliderAutoplayButton.setAttribute('aria-label', 'Pause slideshow');
+    this.autoplay = setInterval(this.autoRotateSlides.bind(this), this.autoplaySpeed);
+  }
+
+  pause() {
+    if (this.isPlaying = false) return;
```

Comment:
```suggestion
    if (!this.isPlaying) return;
```

Also: do we need this condition at all on both `play()` and `pause()`? Isn't it being controlled by the other conditions outside this function?